packages/typespec-test/test/openai_modular/generated/typespec-ts/src/api/operations.ts

```diff
@@ -246,7 +547,9 @@ export function _getChatCompletionsSend(
     .post({
       ...operationOptionsToRequestParameters(options),
       body: {
-        messages: body.messages as any,
+        messages: body["messages"].map((p) =>
+          serializeChatRequestMessageUnion(p)
```

Comment:
This is the place we call the serialization for `ChatRequestMessage[]` which element type is a polymorphic base.